src/Pollers/DeviceMappingPoller.php

```diff
@@ -154,63 +154,106 @@ public function poll(array $work):array
      */
     private function getDeviceMapper(array $hostWithDeviceType, Device $device)
     {
-        switch ($hostWithDeviceType['type_query_result'])
-        {
-            case "1.3.6.1.4.1.161.19.250.256":
+        $oidToMapper = [
+            "1.3.6.1.4.1.161" => "CANOPY",
+            "1.3.6.1.4.1.2736.1.1" => "ETHERWAN",
+            "1.3.6.1.4.1.10002.1" => "UBNT",
+            "1.3.6.1.4.1.14988.1" => "MIKROTIK",
+            "1.3.6.1.4.1.17713.5" => "CAMBIUMPTP500",
+            "1.3.6.1.4.1.17713.6" => "CAMBIUMPTP600",
+            "1.3.6.1.4.1.17713.7" => "CAMBIUMPTP650",
+            "1.3.6.1.4.1.17713.8" => "CAMBIUMPTP800",
+            "1.3.6.1.4.1.17713.9" => "CAMBIUMPTP700",
+            "1.3.6.1.4.1.17713.11" => "CAMBIUMPTP670",
+            "1.3.6.1.4.1.17713.21" => "CAMBIUMEPMP",
+            "1.3.6.1.4.1.17713.250" => "CAMBIUMPTP250",
+            "1.3.6.1.4.1.41112.1.4" => "UBNTAIRMAX",
+            "1.3.6.1.4.1.43356.1.1.1" => "MIMOSABX",
+            "1.3.6.1.4.1.43356.1.1.2" => "MIMOSABX",
+            "1.3.6.1.4.1.43356.1.1.3" => "MIMOSAAX",
+        ];
+
+        $longestOid = 0;
+        $m = "GENERIC";
+
+        foreach ($oidToMapper as $enumeration => $mapperName) {
```

Comment:
I'll merge this in, but something you could do here to not have to loop the entire list every time is first check if the sysObjectID exists as a key in the array (`if (isset($oidMapper[$sysObjectID])) {`) so that you don't have to check the entire thing.

The array value could also just be the mapper itself, so you don't have to switch afterwards.